packages/react/src/products/hooks/useProductDetails.ts

```diff
@@ -0,0 +1,173 @@
+ * @category Products
+ * @subcategory Hooks
+ */
+import {
+  fetchProductDetails as fetchProductDetailsAction,
+  getAllProductSizesRemainingQuantity,
+  getProductBreadcrumbs,
+  getProductError,
+  getProductGroupedEntries,
+  isProductDuplicated,
+  isProductFetched,
+  isProductHydrated,
+  isProductLoading,
+} from '@farfetch/blackout-redux/products';
+import {
+  getProduct,
+  getProductLabelsByPriority,
+  getProductPromotions,
+  isProductOneSize,
+  isProductOutOfStock,
+} from '@farfetch/blackout-redux/entities';
+import { isProductInBag } from '@farfetch/blackout-redux/bags';
+import { useAction } from '../../helpers';
+import { useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import type { StoreState } from '@farfetch/blackout-redux/src/types';
+import type { UseProductDetails } from './types';
+
+/**
+ * Provides state access for dealing with product details business logic.
+ * Also it fetches product details if there are none.
+ *
+ * @param {number} id - Product id to work on.
+ *
+ * @memberof module:products/hooks
+ * @returns {object} All the state needed to manage any product details information.
+ */
+const useProductDetails: UseProductDetails = id => {
+  const fetchProductDetails = useAction(fetchProductDetailsAction);
+  const breadcrumbs = useSelector((state: StoreState) =>
+    getProductBreadcrumbs(state, id),
+  );
+  const error = useSelector((state: StoreState) => getProductError(state, id));
+  const groupedEntries = useSelector((state: StoreState) =>
+    getProductGroupedEntries(state, id),
+  );
+  const isDuplicated = useSelector((state: StoreState) =>
+    isProductDuplicated(state, id),
+  );
+  const isFetched = useSelector((state: StoreState) =>
+    isProductFetched(state, id),
+  );
+  const isHydrated = useSelector((state: StoreState) =>
+    isProductHydrated(state, id),
+  );
+  const isInBag = useSelector((state: StoreState) => isProductInBag(state, id));
+  const isLoading = useSelector((state: StoreState) =>
+    isProductLoading(state, id),
+  );
+  const isOneSize = useSelector((state: StoreState) =>
+    isProductOneSize(state, id),
+  );
+  const isOutOfStock = useSelector((state: StoreState) =>
+    isProductOutOfStock(state, id),
+  );
+  const labelsPrioritized = useSelector((state: StoreState) =>
+    getProductLabelsByPriority(state, id),
+  );
+  const product = useSelector((state: StoreState) => getProduct(state, id));
+  const promotions = useSelector((state: StoreState) =>
+    getProductPromotions(state, id),
+  );
+  const availableSizes = useSelector((state: StoreState) =>
+    getAllProductSizesRemainingQuantity(state, id),
+  );
+
+  useEffect(() => {
+    !isFetched && fetchProductDetails(id);
+  }, [fetchProductDetails, id, isFetched]);
+
+  return {
+    /**
+     * The available sizes for the given product.
+     *
+     * @type {Array}
+     */
+    availableSizes,
+    /**
+     * Product breadcrumbs.
+     *
+     * @type {Array|undefined}
```

Comment:
These types are not exactly matching the ones defined by `UseProductDetails` type.
Also, I believe the @type tag is unnecessary as the type information is already available in typescript, so you can remove the tag (not the comment).

Reply:
Yes they are not exactly matching but this is for jsdoc purposes not typescript

Reply:
I understand but when you have typescript, typings should not be a concern of the jsdoc. We will have to migrate to tsdoc in the future which looks something like this: https://tsdoc.org/

Anyway, at least the same types could be used between jsdoc and `UseProductDetails` type, so there aren't any mismatches if someone changes the type in Typescript and forgets to change the type in the jsdoc.

For the breadcrumbs property it could be:

```javascript
/**
     * Product breadcrumbs.
     *
     * @type {BreadCrumb[] | undefined}
```

Reply:
Its a good idea to have a more integrated documentation with typescript, but I think that its better to change documentation after changing the technology :)

Reply:
Ok, I will close this but keep in mind that the types between jsdoc and `UseProductDetails` can be out-of-sync.